### DISCUSSION TITLE
Clarification on error codes for MediaCodec API

### DISCUSSION THREAD

**Discussion by zmerp** [android/ndk#1768] [Q&A]
Hi, I would like to know more precisely which error/info codes can be returned by `AMediaCodec_dequeueInputBuffer` and `AMediaCodec_dequeueOutputBuffer`. In particular I know that they can return both positive and negative error codes, the two positive error codes being `AMEDIACODEC_ERROR_INSUFFICIENT_RESOURCE` and `AMEDIACODEC_ERROR_RECLAIMED`, what would be a maximum valid buffer index or minimum positive error code that is guaranteed to hold in a forward-compatible way? And likewise, is there a minimum info code?

Related issue: https://github.com/rust-windowing/android-ndk-rs/pull/316#issuecomment-1240605765

**Comment by DanAlbert** [comment]
You'll have to file a platform bug. We're just the messengers for the platform APIs. You already know more about the possible errors codes than we do.

**Reply by MarijnS95** [reply]
@DanAlbert And where do we file these "platform bugs" or "requests for more detailed documentation"? Somewhere on https://r.android.com?

**Reply by DanAlbert** [reply]
https://b.android.com

**Reply by DanAlbert** [reply]
Now that I have a moment to answer in more detail (sorry for being brief earlier, I had just a second between meetings and figured any answer was better than a delayed answer), I think you want the "media" component listed there? I frankly don't know and have a hard time dealing with that page myself, I just know that I _don't_ have the answer. Even if you misfile there, it should hopefully get re-triaged. If it doesn't, paste it here and we'll go prod someone.

**Reply by MarijnS95** [reply]
@zarik5 Have you created a bug report at https://issuetracker.google.com/issues/new?component=898361&template=1458029 yet? I'd love to clear out some NDK pull requests :)

**Reply by zmerp** [reply]
@MarijnS95 thanks for the reminder. I opened an issue: https://issuetracker.google.com/u/1/issues/255698045
